### DISCUSSION TITLE
error when i download `npm install --global hardhat-shorthand`

### DISCUSSION THREAD

**Discussion by TimeKeyRoll** [NomicFoundation/hardhat#3107] [Troubleshooting]
I tried to download this `npm install --global hardhat-shorthand` while running `vscode` as `adminstartive` but getting this error.

```

PS D:\hardhat-lottery-fcc> npm install --global hardhat-shorthand
npm ERR! code EPERM
npm ERR! syscall mkdir
npm ERR! path C:\Program Files\nodejs\node_modules\.hardhat-shorthand-mjYILpcy
npm ERR! errno -4048
npm ERR! Error: EPERM: operation not permitted, mkdir 'C:\Program Files\nodejs\node_modules\.hardhat-shorthand-mjYILpcy'
npm ERR!  [Error: EPERM: operation not permitted, mkdir 'C:\Program Files\nodejs\node_modules\.hardhat-shorthand-mjYILpcy'] {
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: 'C:\\Program Files\\nodejs\\node_modules\\.hardhat-shorthand-mjYILpcy'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It's possible that the file was already in use (by a text editor or antivirus),
npm ERR! or that you lack permissions to access it.
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
```

Anyone knows about this?

**Comment by fvictorio** [comment]
Can you install other global packages, or is your issue that you can't install any global package? If it's the latter, that depends a lot on your OS and how you have node installed.

**Reply by TimeKeyRoll** [reply]
And I don't remember downloading 'anything  global' but I can try. Can you tell me what should I try to download as global of it's with OS problem or with that specific. Thanks

**Reply by fvictorio** [reply]
You have to install it with the `--global` flag: `npm i --global hardhat-shorthand`

**Reply by TimeKeyRoll** [reply]
Absolutely that was my first question.
That's what I did and that is what got me stuck there with an error which is mentioned above.

So how do I fix the error?

**Reply by fvictorio** [reply]
Sorry, I see how my comment wasn't helpful. My question is: if you run other global package, does it work?

For example, try this:

```
npm i -g create-react-app
create-react-app --help
```

**Reply by TimeKeyRoll** [reply]
It has the same error as previous one.
So it means it's issue with the OS or windows(7) I'm using?

Do you have any idea?

**Reply by fvictorio** [reply]
Yes, it has something to do with windows and your npm installation. Sadly, I can't help you with that.

**Reply by TimeKeyRoll** [reply]
OK. Thanks
